e2e-tests/gef/cypress/e2e/amend-facility.spec.js

```diff
@@ -0,0 +1,45 @@
+import relative from './relativeURL';
+import CREDENTIALS from '../fixtures/credentials.json';
+import CONSTANTS from '../fixtures/constants';
+import { MOCK_USER_MAKER } from '../fixtures/mocks/mock-user-maker';
+import { MOCK_APPLICATION_AIN } from '../fixtures/mocks/mock-deals';
+import { MOCK_FACILITY_ONE } from '../fixtures/mocks/mock-facilities';
+import applicationPreview from './pages/application-preview';
+
+let acknowledgedDealId;
+let token;
+
+context('Amend a facility', () => {
+  before(() => {
+    cy.apiLogin(CREDENTIALS.MAKER).then((t) => {
+      token = t;
+    }).then(() => {
+      cy.apiCreateApplication(MOCK_USER_MAKER, token).then(({ body }) => {
+        acknowledgedDealId = body._id;
+        cy.apiUpdateApplication(acknowledgedDealId, token, MOCK_APPLICATION_AIN).then(() => {
+          cy.apiCreateFacility(acknowledgedDealId, CONSTANTS.FACILITY_TYPE.CASH, token).then((facility) => {
+            MOCK_FACILITY_ONE._id = facility.body.details._id;
+            cy.apiUpdateFacility(facility.body.details._id, token, MOCK_FACILITY_ONE);
+          });
+          cy.apiSetApplicationStatus(acknowledgedDealId, token, CONSTANTS.DEAL_STATUS.UKEF_ACKNOWLEDGED);
+        });
+      });
+    });
+  });
+
+  beforeEach(() => {
+    cy.saveSession();
+    cy.login(CREDENTIALS.MAKER);
+  });
+
+  describe('Amend a facility', () => {
```

Comment:
I'm not sure if we do much 'negative' testing but do we also need a test for making sure other roles can't see the button? (It's covered a bit in API tests but not here - over to you again @abhi-markan for current conventions on the codebase/e2e tests)

Reply:
I agree @ChristianMcc.
@JennyHumphries  would be nice to add some assertions covering following:
* Login as a maker from different bank (thought already covered in unit test).
* Login as a checker from the same bank.
* Facility with status `un-issued`.
* Deal with a different stage from `Acknowledged` please.

Reply:
No problem, I can add those test cases.

What should the behaviour be for a facility that's unissued? I think this is something we'll need to add to the ticket ACs

Reply:
I've added test cases to show the button is hidden when the deal status is not acknowledged or if logged in as a checker.

I haven't added one for if the user a maker from a different bank because I think they get blocked from seeing the whole application-preview page so it doesn't really work for a cypress test. This case is covered in the unit tests.

I'll add one for the un-issued facility once I've verified what the expected behaviour is.

Reply:
Hello @JennyHumphries apologies I have just seen your question. Expected behaviour for un-issued facility is that we do not see the button since amendment can only be applied on `issued` facility.

Reply:
No worries! I've updated to add this condition.